### PR TITLE
Update for OAuth (switch to samwilson/phpflickr library)

### PR DESCRIPTION
**Pull request by samwilson**
This switches to a fork of the PhpFlickr library that supports
OAuth, and makes related changes required to work with that.

It also adds a feature by which it stores the access token
credentials in site config, so that the user doesn't have to
re-authenticate every time.

Refs #5

### CODE REVIEW

**Comment by mistic100** [comment]
You must commit the `vendor` directory. Piwigo does not work with composer so everything must be in the repository when publishing the plugin from piwigo.org/ext

File: admin/import.php [hunk 1]
```diff
@@ -187,8 +207,8 @@
 
     foreach ($page_photos as &$photo)
     {
-      $photo['thumb'] = $flickr->buildPhotoURL($photo, "thumbnail");
-      $photo['src'] = $flickr->get_biggest_size($photo['id'], "medium_800");
+      $photo['thumb'] = $photo['url_t'];
+      $photo['src'] = $photo['url_m'];
```
Comment (mistic100) [inline admin/import.php]: It is important to try to get the biggest size available like  I did not just the medium.

Comment (samwilson) [reply admin/import.php]: This `src` is only used for the 'zoom' function when previewing the photos that are to be imported. It doesn't determine what's actually imported (that's in [ws_functions.php](https://github.com/mistic100/Flickr2Piwigo/pull/7/files#diff-3e3297a9d9faa54640a8e0d574c5ae5cR57)).

The reason I switched it to these image URLs is that they can be retrieved in the `photosets.getPhotos` Flickr API call, and so save on a request.

**Comment by samwilson** [comment]
Thanks for reviewing so quickly!

I was wondering about the vendor directory. I'm not at all familiar with the packaging process for uploading to piwigo.org.

I'll add it to this patch.

**Comment by samwilson** [comment]
Vendor now committed; ready for re-review. Thanks!

**Comment by samwilson** [comment]
Actually, isn't it possible to use the 'Download from URL' option of the Piwigo extension system, and point it to a prepared release on Github? That way, we don't have to commit vendor (which would be rather preferable I reckon).

File: admin/import.php (hunk 1 above)
Comment (mistic100) [reply admin/import.php]: Ok I forgot that

**Comment by mistic100** [comment]
You didn't committed the vendor folder.... problably because it's still in the .gitignore

Yes we could use this feature but that means the release process is not automatic anymore, no replacement of the "Version" in the main file, same thing for "Plugin URI".

**Comment by samwilson** [comment]
No I did, but then I undid it because I wanted to see what you thought about the tarball upload idea.

But you're right, the whole thing is simpler if it's committed. Done so now.

**Comment by mistic100** [comment]
Great, now PHP looks like a modern language with hundreds of files from too many dependencies (well it will always be bellow a big Node application requiring at least 1500 packages !)

Is it ready for publish ? I cannot test it.
Or as suggested in #5 I transfer the repo to you :-)

**Comment by samwilson** [comment]
hehe yup! it's not "enterprise ready" until we're 1000 files deep I guess. :-)

And yes, I'm happy to take over maintaining it if you like. We can be co-maintainers yes? The plugin system allows that?

No need to transfer the repo though, probably better to switch the URLS, so that it's obvious there's been a change.

**Comment by mistic100** [comment]
Yes it's simplier, are you "samwilson" on piwigo.org too ?

**Comment by samwilson** [comment]
Yep.

**Comment by mistic100** [comment]
Done, I'm really happy someone takes the maintenance.
25 more plugins to go :-)

**Comment by samwilson** [comment]
Cool, thanks!

So I'll merge this, and then tag 1.3.0?

Nothing is considered "released" until a new revision is created on piwigo.org is it?

**Comment by mistic100** [comment]
That's right.

**Pull request merged**